crates/sui-indexer-alt/src/ingestion/remote_client.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ingestion::client::{FetchError, FetchResult, IngestionClientTrait};
+use crate::ingestion::remote_client::HttpError::Http;
+use crate::ingestion::Result as IngestionResult;
+use reqwest::{Client, StatusCode};
+use tracing::{debug, error};
+use url::Url;
+
+pub(crate) struct RemoteIngestionClient {
+    url: Url,
+    client: Client,
+}
+
+impl RemoteIngestionClient {
+    pub(crate) fn new(url: Url) -> IngestionResult<Self> {
+        Ok(Self {
+            url,
+            client: Client::builder().build()?,
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl IngestionClientTrait for RemoteIngestionClient {
+    /// Fetch a checkpoint from the remote store.
+    ///
+    /// Transient errors include:
+    ///
+    /// - failures to issue a request, (network errors, redirect issues, etc)
+    /// - request timeouts,
+    /// - rate limiting,
+    /// - server errors (5xx),
+    /// - issues getting a full response and deserializing it as [CheckpointData].
```

Comment:
nit: this transient error is handled by `IngestionClient` still, IIUC, so we should update its doc comment to reflect that and remove it from here.